pkgs/java_http/lib/src/java_client.dart

```diff
@@ -28,16 +33,109 @@ class JavaClient extends BaseClient {
     // See https://github.com/dart-lang/http/pull/980#discussion_r1253700470.
     _initJVM();
 
-    final javaUrl = URL.ctor3(request.url.toString().toJString());
-    final dartUrl = Uri.parse(javaUrl.toString1().toDartString());
+    final (statusCode, reasonPhrase, responseHeaders, responseBody) =
+        await Isolate.run(() {
+      request.finalize();
 
-    const result = 'Hello World!';
-    final stream = Stream.value(latin1.encode(result));
+      final httpUrlConnection = URL
+          .ctor3(request.url.toString().toJString())
+          .openConnection()
+          .castTo(HttpURLConnection.type, deleteOriginal: true);
 
-    return StreamedResponse(stream, 200,
-        contentLength: 12,
-        request: Request(request.method, dartUrl),
-        headers: {'content-type': 'text/plain'},
-        reasonPhrase: 'OK');
+      request.headers.forEach((headerName, headerValue) {
+        httpUrlConnection.setRequestProperty(
+            headerName.toJString(), headerValue.toJString());
+      });
+
+      httpUrlConnection.setRequestMethod(request.method.toJString());
+
+      final statusCode = _statusCode(request, httpUrlConnection);
+      final reasonPhrase = _reasonPhrase(httpUrlConnection);
+      final responseHeaders = _responseHeaders(httpUrlConnection);
+      final responseBody = _responseBody(httpUrlConnection);
+
+      httpUrlConnection.disconnect();
+
+      return (
+        statusCode,
+        reasonPhrase,
+        responseHeaders,
+        responseBody,
+      );
+    });
+
+    final contentLengthHeader = responseHeaders['content-length'];
+    final contentLength = (contentLengthHeader == null)
+        ? null
+        : int.tryParse(contentLengthHeader);
```

Comment:
I would `throw` if the header isn't an integer.

Reply:
I've pushed a commit which changes it to :
```dart 
int.parse(contentLengthHeader);
```
this should throw a `FormatException` if the header isn't an integer.

### Alternative Solution 1
We could catch the `FormatException` and throw a `ClientException` with a more detailed message:
```dart
int? contentLength;
try {
  contentLength =
      (contentLengthHeader == null) ? null : int.parse(contentLengthHeader);
} on FormatException {
  throw ClientException(
      'Invalid content-length header: $contentLengthHeader.'
      ' Content-length must be a non-negative integer.',
      request.url);
}
```

### Alternative Solution 2
We could use the [`getContentLengthLong()`](https://docs.oracle.com/javase/8/docs/api/java/net/URLConnection.html#getContentLengthLong--) method in `URLConnection`, which returns -1 if the content length is not known:

```dart
int? contentLength = httpUrlConnection.getContentLengthLong();
contentLength = (contentLength == -1) ? null : contentLength;
```